src/main/java/org/folio/circulation/domain/validation/overriding/OverrideValidator.java

```diff
@@ -0,0 +1,42 @@
+package org.folio.circulation.domain.validation.overriding;
+
+import static org.folio.circulation.domain.LoanAction.CHECKED_OUT_THROUGH_OVERRIDE;
+import static org.folio.circulation.support.ValidationErrorFailure.singleValidationError;
+import static org.folio.circulation.support.results.Result.ofAsync;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.support.http.server.InsufficientOverridePermissionsError;
+import org.folio.circulation.support.results.Result;
+
+public abstract class OverrideValidator implements LoanValidator {
+  protected static final String OKAPI_PERMISSIONS = "x-okapi-permissions";
+  private final String comment;
+
+  public OverrideValidator(String comment) {
+    this.comment = comment;
+  }
+
+  @Override
+  public CompletableFuture<Result<LoanAndRelatedRecords>> validate(LoanAndRelatedRecords records) {
+    return ofAsync(() -> records)
+      .thenCompose(result -> result.failAfter(relatedRecords -> isOverridingForbidden(),
+        relatedRecords -> singleValidationError(
+          new InsufficientOverridePermissionsError(null, null))))
```

Comment:
We'll need to handle this error here with the appropriate type (INSUFFICIENT_OVERRIDE_PERMISSIONS)